dotCMS/src/main/java/com/dotmarketing/osgi/GenericBundleActivator.java

```diff
@@ -814,38 +836,16 @@ protected void unregisterServices ( BundleContext context ) throws Exception {
      */
     protected void unpublishBundleServices () throws Exception {
 
-        if ( this.context == null ) {
-            this.context = context;
-        }
-
-        //ClassLoaders
-        ClassLoader felixClassLoader = getFelixClassLoader();
-        ClassLoader contextClassLoader = getContextClassLoader();
-
-        //Force the loading of some classes that may be already loaded on the host classpath but we want to override with the ones on this bundle
-        String overrideClasses = getManifestHeaderValue( context, MANIFEST_HEADER_OVERRIDE_CLASSES );
-        if ( overrideClasses != null && !overrideClasses.isEmpty() ) {
-
-            String[] forceOverride = overrideClasses.split( "," );
-            if ( forceOverride.length > 0 ) {
-
+        if (null != this.overriddenClasses && !this.overriddenClasses.isEmpty()) {
+            for (String overridenClass : this.overriddenClasses) {
                 try {
-                    //Get the activator class for this OSGI bundle
-                    String activatorClass = getManifestHeaderValue( context, MANIFEST_HEADER_BUNDLE_ACTIVATOR );
-                    //Injecting this bundle context code inside the dotCMS context
-                    addClassTodotCMSClassLoader( activatorClass );
-                } catch ( Exception e ) {
-                    Logger.error( this, "Error injecting context for overriding", e );
-                    throw e;
-                }
-
-                for ( String classToOverride : forceOverride ) {
-                    ByteBuddyAgent.install();
-                    new ByteBuddy()
-                            .rebase(Class.forName(classToOverride.trim()), ClassFileLocator.ForClassLoader.of(contextClassLoader))
-                            .name(classToOverride.trim())
-                            .make()
-                            .load(contextClassLoader,ClassReloadingStrategy.fromInstalledAgent());
+                    this.classReloadingStrategy
+                            .reset(ClassFileLocator.ForClassLoader.of(getContextClassLoader()),
+                                    Class.forName(overridenClass));
+                } catch (Exception e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)